home/serializers_v3.py

```diff
@@ -0,0 +1,433 @@
+                        ),
+                        (
+                            "example_values",
+                            format_example_values(template.example_values.raw_data),
+                        ),
+                        ("category", template.category),
+                        ("submission_name", template.submission_name),
+                        ("submission_status", template.submission_status),
+                        ("submission_result", template.submission_result),
+                    ]
+                )
+            )
+
+        elif str(block.block_type) == "Whatsapp_Message":
+            message = block.value
+
+            image = message["image"]
+            image = image.id if image is not None else None
+
+            messages.append(
+                OrderedDict(
+                    [
+                        ("type", block.block_type),
+                        ("image", image),
+                        ("media", message["media"]),
+                        ("document", message["document"]),
+                        ("text", message["message"]),
+                        (
+                            "buttons",
+                            format_buttons_and_list_items(message["buttons"].raw_data),
+                        ),
+                        (
+                            "list_items",
+                            format_buttons_and_list_items(
+                                message["list_items"].raw_data
+                            ),
+                        ),
+                        ("list_title", message["list_title"]),
+                        (
+                            "variation_messages",
+                            format_variation_messages(message["variation_messages"]),
+                        ),
+                    ]
+                )
+            )
+        else:
+            raise Exception("Unknown Block Type detected")
+
+    return messages
+
+
+class ContentPageSerializerV3(PageSerializer):
+    title = serializers.CharField(read_only=True)
+    slug = serializers.SlugField(read_only=True)
+    messages = serializers.SerializerMethodField()
+    detail_url = serializers.SerializerMethodField()
+    meta_fields = []
+
+    class Meta:
+        model = ContentPage
+
+        fields = [
+            "slug",
+            "detail_url",
+            "locale",
+            "title",
+            "subtitle",
+            "messages",
+            "tags",
+            "triggers",
+            "has_children",
+            "related_pages",
+        ]
+
+        # exclude = ["meta_fields"]
+
+    def get_messages(self, obj):
+        request = self.context["request"]
+        messages = body_field_representation_V3(obj, request)
+        return messages
+
+    def get_buttons(self, obj):
+        buttons = list(obj.buttons.raw_data)
+        for button in buttons:
+            if "id" in button:
+                del button["id"]
+        return buttons
```

Comment:
Can we not use the `format_buttons_and_list_items` function here? Also, I don't think we do buttons or example values directly here, so we can probably remove these two functions? (they're not in the fields list)

Reply:
Done